main.py

```diff
@@ -89,7 +89,42 @@ def shortest_path(self):
         paths = self._filter_paths(paths, desired, undesired)
         return jsonify({'paths': paths})
 
-    @listen_to('kytos.topology.updated')
+    @rest('v3/', methods=['POST'])
+    def shortest_constrained_path(self):
+        """Get the set of shortest paths between the source and destination."""
+        data = request.get_json()
+
+        source = data.get('source')
+        destination = data.get('destination')
+        base_metrics = data.get('base_metrics', {})
+        try:
+            paths = self.graph.constrained_flexible_paths(source,
+                                                          destination,
+                                                          base=base_metrics)
+            return jsonify(paths)
+        except TypeError as err:
+            return jsonify({"error": err})
+
+    @ rest('v4/', methods=['POST'])
+    def shortest_constrained_path2(self):
+        """Get the set of shortest paths between the source and destination."""
+        data = request.get_json()
+
+        source = data.get('source')
+        destination = data.get('destination')
+        base_metrics = data.get('base_metrics', {})
+        fle_metrics = data.get('flexible_metrics', {})
+        depth = data.get('depth', 0)
+        try:
+            paths = self.graph.constrained_flexible_paths(source, destination,
+                                                          depth,
+                                                          base=base_metrics,
+                                                          flexible=fle_metrics)
+            return jsonify(paths)
+        except TypeError as err:
+            return jsonify({"error": err})
```

Comment:
We added `shortest_constrained_path2` after our product owner suggested that we also let the user select flexible and inflexible metrics. So to implement the alternate design in a timely fashion, we created that method to avoid possibly affecting the implementation of our original design.

But at this point, both methods can be consolidated into one.